### PR TITLE
Fix deformable convolution cl kernel for Multi-Groups

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Update for-loop and fix index operations*
 - *Add Multi-Groups testcases*

### Tickets:
 - *59600*

### CODE REVIEW

**Comment by steve-y** [comment]
@yeonbok  @isanghao Could you review this PR? This PR added Group = 2 test case to gpuFuncTests like cpuFuncTests, as cpuFuncTests has already has it.

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl [hunk 1]
```diff
@@ -48,85 +48,90 @@ KERNEL(deformable_convolution_gpu_bfyx_ref)(
...
-        const int trans_offset = b * INPUT1_BATCH_PITCH + 2 * dg_offset;
-#if DEFORMABLE_MASK_ENABLED
-        const int mask_offset = b * INPUT2_BATCH_PITCH + dg_offset;
+    const int offset_size = FILTER_SIZE_Y * FILTER_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_X;
+
+    for (uint c = 0; c < FILTER_IFM_NUM; ++c) {
+#if GROUPED
+        const int deformable_group_idx = (FILTER_IFM_NUM * of + c) / (FILTER_IFM_NUM * FILTER_GROUPS_NUM / DEFORMABLE_GROUPS);
+#else
+        const int deformable_group_idx = (FILTER_IFM_NUM * of + c) / (FILTER_IFM_NUM * FILTER_GROUPS_NUM / DEFORMABLE_GROUPS) % DEFORMABLE_GROUPS;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl:57]: I am not familiar with this kernel but original impl was handling batch in the offset while the new impl is not. Could you explain whats the difference?

Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl:57]: It is implemented by referring to `src/core/reference/include/ngraph/runtime/reference/deformable_convolution.hpp` and batch is not used in it to calculate `deformable_group_idx`. So the batch is not used in this code.

Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl:57]: Added test cases of batch = 2, 3, 4 to group/deformable_group = 2/2, 4/1, 4/2.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**